File: system-test/java/src/main/java/com/optivem/shop/testkit/dsl/port/given/steps/GivenCountry.java
package com.optivem.shop.testkit.dsl.port.given.steps;

import com.optivem.shop.testkit.dsl.port.given.steps.base.GivenStep;

public interface GivenCountry extends GivenStep {
    GivenCountry withCode(String country);

    GivenCountry withTaxRate(String taxRate);

    GivenCountry withTaxRate(double taxRate);
}

File: system-test/typescript/src/testkit/dsl/port/given/steps/given-country.ts
import type { GivenStage } from '../given-stage.js';
import type { WhenStage } from '../../when/when-stage.js';
import type { ThenStage } from '../../then/then-stage.js';

export interface GivenCountry {
  withCode(country: string): this;
  withTaxRate(taxRate: string | number): this;
  and(): GivenStage;
  when(): WhenStage;
  then(): ThenStage;
}
